src/core/zowe/core_for_zowe_sdk/credential_manager.py

```diff
@@ -0,0 +1,195 @@
+        None
+        """
+        
+        try:
+            keyring.delete_password(service_name, account_name)  
+        except keyring.errors.PasswordDeleteError:
+            # Handling multiple credentials stored when the operating system is Windows
+            if sys.platform == "win32":
+                # Delete the secure value with an index
+                index = 1
+                while True:
+                    field_name = f"{account_name}-{index}"
+                    try:
+                        keyring.delete_password(f"{service_name}-{index}", field_name)
+                    except keyring.errors.PasswordDeleteError:
+                        break
+                    index += 1
+    
+    @staticmethod
+    def save_secure_props()-> None:
+        """
+        Set secure_props for the given config file
+        Returns
+        -------
+        None
+        """
+        if not HAS_KEYRING:
+            return
+        
+        # Filter or suppress specific warning messages
+        warnings.filterwarnings("ignore", message="^Retrieved an UTF-8 encoded credential")
+        service_name = constants["ZoweServiceName"]
+        credential =  CredentialManager.secure_props
+        # Check if credential is a non-empty string
+        if credential:
+            is_win32 = sys.platform == "win32"
+            #  = "UTF-16" if is_win32 else "UTF-8"
+            if is_win32:
+                service_name += "/" + constants["ZoweAccountName"] 
+            
+            # Load existing credentials, if any
+            existing_credential = CredentialManager._retrieve_credential(constants["ZoweServiceName"])
+            if existing_credential:
+                    
+                # Decode the existing credential and update secure_props
+                existing_credential_bytes = base64.b64decode(existing_credential).decode()
+                existing_secure_props = commentjson.loads(existing_credential_bytes)
+                existing_secure_props.update(credential)
+                # Encode the credential
+                encoded_credential = base64.b64encode(commentjson.dumps(existing_secure_props).encode()).decode()
+                # Delete the existing credential
+                CredentialManager.delete_credential(service_name , constants["ZoweAccountName"])
+            else:
+                # Encode the credential
+                encoded_credential = base64.b64encode(commentjson.dumps(credential).encode()).decode() 
+            # Check if the encoded credential exceeds the maximum length for win32
+            if is_win32 and len(encoded_credential) > constants["WIN32_CRED_MAX_STRING_LENGTH"]:
+                # Split the encoded credential string into chunks of maximum length
+                chunk_size = constants["WIN32_CRED_MAX_STRING_LENGTH"]
+                chunks = [encoded_credential[i: i + chunk_size] for i in range(0, len(encoded_credential), chunk_size)]
+                chunks[-1]+= '\0'
```

Comment:
If the password is 5120 (2560*2) bytes, this will make the last chunk 2561 bytes which causes an error.
This can be fixed by adding the trailing `\0` byte before splitting the password in chunks instead of after.